### ISSUE TITLE
Gladiatus Update 4.3.1 Bugs

### ISSUE THREAD

**Issue by FrutyX** [DinoDevs/GladiatusCrazyAddon#382]
Gameforge moved their images to a new URL location, so images GCA is using are not being loaded and that is causing a mess.

## ~~Reports are failing to detect enemies URL:~~
🟢 **Status:** FIXED

<details>

`Failed to detect enemy url("//gf3.geo.gfsrv.net/cdn29/bac7e539d61a9de6caec21363c2694.jpg")`

</details>

## ~~Mystery box rubies prices unknown:~~
🟢 **Status:** FIXED
<details>

![rubiesprices](https://user-images.githubusercontent.com/23278552/180775996-c353061a-8192-4580-a43e-dac693b2164b.PNG)

</details>

## ~~Pantheon quests bugged:~~
🟢 **Status**: FIXED

<details>

```
pantheon.quests.js?4.3.6&built=1658150383374:82 Uncaught TypeError: Cannot read properties of null (reading '1')
    at Object.quests_reorder (pantheon.quests.js?4.3.6&built=1658150383374:82:122)
    at Object.inject (pantheon.quests.js?4.3.6&built=1658150383374:15:9)
    at fireLoad (pantheon.quests.js?4.3.6&built=1658150383374:180:23)
```
</details>

## ~~Guild battle report bugged:~~
🟢 **Status:** FIXED

<details>

![guildmsg](https://user-images.githubusercontent.com/23278552/179835127-1ccbd308-5c80-4614-924e-89493cb45435.PNG)

</details>

## ~~Add to Target List bugged:~~
🟢 **Status:** FIXED 

<details>

![targetlist](https://user-images.githubusercontent.com/23278552/179520794-526ad0e3-c819-45b5-a17f-7346cc077ca0.PNG)

</details>

## ~~Gold&XP Statistics Icon bugged:~~
🟢 **Status:** FIXED

<details>

![statistics](https://user-images.githubusercontent.com/23278552/179521020-cdfd4800-66d5-43dc-b0a7-c5c7300332f4.PNG)

</details>

## ~~Negotium X enemies bugged:~~
🟢 **Status:** FIXED

<details>

![nego](https://user-images.githubusercontent.com/23278552/179521198-aa849e16-f5cc-4538-aba7-787869503dd5.PNG)

</details>

## ~~Buffs bugged when viewing other player profiles:~~
🟢 **Status:** FIXED

<details>

![buffs](https://user-images.githubusercontent.com/23278552/179521951-a67a8b04-8ed6-4b03-8306-33b724d715c0.PNG)

</details>

## ~~CT and Dungeons table overview bugged:~~
🟢 **Status:** FIXED

<details>

![ct](https://user-images.githubusercontent.com/23278552/179524567-f0e90b05-8ecf-4a88-ad6b-fffdbf6350f1.PNG)

```
Failed to load resource: the server responded with a status of 404 (Not Found)
char_status_kopf_b.jpg:1
Failed to load resource: the server responded with a status of 404 (Not Found)
char_status_abschluss_b.jpg:1
```
</details>

## ~~Durability loss numbers are bugged with GCA enabled:~~
🟢 **Status:** FIXED

<details>

![numbersbug](https://user-images.githubusercontent.com/23278552/179643570-05403b5d-9be8-4c6c-af71-0a3bec7b5e57.png)

</details>

## ~~Attack guild button missing in War Master Hall:~~
🟢 **Status:** FIXED

<details>

![attackbutton](https://user-images.githubusercontent.com/23278552/179741576-0d910779-12d8-40e9-895b-53469368ff1c.PNG)

</details>

## ~~Library level indicator bugged:~~
🟢 **Status:** FIXED

<details>

![librarylevels](https://user-images.githubusercontent.com/23278552/180639241-d78d56f9-9f79-498a-b2ac-ac1a26070396.PNG)

</details>

##
... more will be added when found.

**Comment by GramThanos** [comment]
Sum up:

Show enemy drops was disabled - (a walk-a-round to identify NPCs is needed)
Reorder quests was disabled - (a walk-a-round to identify quest categories is needed)
Image caching was disabled - (this was a Gladiatus bug fix, we have to check whether the bug was fixed)

**Comment by FrutyX** [comment]
A very quick list update has been made.

**Comment by GramThanos** [comment]
> > Sum up:
> > Show enemy drops was disabled - (a walk-a-round to identify NPCs is needed) Reorder quests was disabled - (a walk-a-round to identify quest categories is needed) Image caching was disabled - (this was a Gladiatus bug fix feature, we have to re script it, the bug is still here)
> 
> For NPCs the fastest way is probably caching the names when you visit each location. For quests, more extreme measures like identifying images by comparing pixels at specific locations of converting the images to base64 may be required (that may be heavy for old PCs).

NPCs > You have to identify the location too
Quests > The images are loaded from an other domain, thus javascript don't have access to their pixels.

**Comment by FrutyX** [comment]
After a while, found a new bug, added on the list.

**Comment by FrutyX** [comment]
According from what I've seen on Glad discord, it appears the "attack other guilds" button is bugged with GCA enabled, can someone confirm this? I can't, no guilds to attack.

**Comment by FrutyX** [comment]
The first merges are coming tomorrow, wouldn't be good to disable Global Arena and the simulator before? Remotely? It will cause havoc in both, because old server will become invalid.

Or do you plan to keep up with it and live update it during the merges? I mean, that's only up to you how you manage it, I just know we don't need another thing that will break the extension.

**Comment by FrutyX** [comment]
Uhhh, what feature was that again, I haven't seen that in a while.

![guildmsg](https://user-images.githubusercontent.com/23278552/179784776-753bee1b-80b0-4de6-a22c-35f7f4ad65be.PNG)

This is supposed to look different right? There was an overview of the battle I believe?

**Comment by GramThanos** [comment]
The addon was loading the report and was showing the battle results in a report like way (attacker vs defender with points).

**Comment by FrutyX** [comment]
Very awesome, Thanos is today's MVP.

**Comment by SleepingShadow-bit** [comment]
Unfortunately I don't see any change in the quest.  Correctly?

**Comment by FrutyX** [comment]
Something wrong on your side, it is fixed, remove the addon completely.

**Comment by SleepingShadow-bit** [comment]
New download.
Data and cache cleared... Grml. I will check again dude

**Comment by GreatApo** [comment]
> New downloaded Data and cache cleared... Grml. I will check again dude

It's just reordering the quests, not too much of a change

**Comment by SleepingShadow-bit** [comment]
Reinstalled twice - no success.  Third time installed - fine.  
Thanks for the fix

**Comment by FrutyX** [comment]
Btw, you did reinstalled, but have you restarted your browser? I want the truth :D

**Comment by SleepingShadow-bit** [comment]
> Btw, you did reinstalled, but have you restarted your browser? I want the truth :D

Hmm good question. I don't think so haha
Its my bad xD

**Comment by GramThanos** [comment]
Probably we will have to move CDN related code to a new script... The resources drop will work now, but not all enemies can be identified.

**Comment by FrutyX** [comment]
Oh god
![location](https://user-images.githubusercontent.com/23278552/180065987-4a851f13-ffd2-48f1-99ea-7f9e484e6138.PNG)

**Comment by FrutyX** [comment]
Yes of course, sorry.

//gf1.geo.gfsrv.net/cdn96/28735e13275819747922c16a516a5b.jpg
//gf3.geo.gfsrv.net/cdn24/d84f6adad4686ae45ff5b98fcbc692.jpg
//gf2.geo.gfsrv.net/cdn13/b173612baddf3d7dadbcdb823d3a19.jpg
//gf3.geo.gfsrv.net/cdn29/bac7e539d61a9de6caec21363c2694.jpg

**Comment by FrutyX** [comment]
Here is some more, I have locked two last Britannia locations unfortunately.

`//gf1.geo.gfsrv.net/cdn9d/d8e4b82f457b0803419ce2c50f5255.jpg
//gf3.geo.gfsrv.net/cdne8/5d44a11052e740a8ec483e76f181a7.jpg
//gf1.geo.gfsrv.net/cdn67/a2611142886c9c6634cbb31e0cdb5e.jpg
//gf1.geo.gfsrv.net/cdn94/30c496661bc490a06b5c212124f4fe.jpg

//gf1.geo.gfsrv.net/cdnc5/8fc1dc28a272c41b475828a8107afe.jpg
//gf3.geo.gfsrv.net/cdn8d/0ead089e64b248786fb3e126ffcf7f.jpg
//gf3.geo.gfsrv.net/cdn58/99db5b8f77050f201cf67dc8e5bf9d.jpg
//gf1.geo.gfsrv.net/cdn63/cdbea7d87f34475e22555256f0f942.jpg

//gf1.geo.gfsrv.net/cdn64/b98041c012f259885ba823c53eeaca.jpg
//gf2.geo.gfsrv.net/cdnab/195c75c0ba9e8fadda41b58b107b84.jpg
//gf1.geo.gfsrv.net/cdn32/3ae0563db2cd7241d349f766002b3b.jpg
//gf1.geo.gfsrv.net/cdncd/9dd0228c3a6c8dfe9e5b9bd20f50a5.jpg

//gf1.geo.gfsrv.net/cdnf3/841d17a434e2018c8c84c1fa2271a5.jpg
//gf1.geo.gfsrv.net/cdn63/e3e4aa1f515a7c28ce7812c3ac63b7.jpg
//gf1.geo.gfsrv.net/cdn04/f4e1cd5cd2438f5a41c47814d0a710.jpg
//gf2.geo.gfsrv.net/cdn1b/18b0984d6a55da3e034e4760628b7f.jpg`

**Comment by GreatApo** [comment]
That should be enough data for that Thanos to crack the rest :P
Alternatively, we will have to gather them by hand.

**Comment by FrutyX** [comment]
By the way, I know this is still work in progress, but with those URLs added, I am still getting those reverse lookup url errors, just letting you know.

**Comment by FrutyX** [comment]
Alright, so now tell me, what is the status of detect enemies bug? Are you gonna improve it or we can call it fixed? We don't have the two last Britannia locations, I don't know if Thanos can get them without accessing them, if not, just tell me and I will put some personal effort, to get those URLs from someone.

**Comment by GreatApo** [comment]
> Alright, so now tell me, what is the status of detect enemies bug? Are you gonna improve it or we can call it fixed? We don't have the two last Britannia locations, I don't know if Thanos can get them without accessing them, if not, just tell me and I will put some personal effort, to get those URLs from someone.

Thanos managed to optimize and get the hash keys for most images, but some of them (the ones commented) are wrong (they probably have a different optimisation and different hashes are generated. We will gradually collect them all, you don't have to gather the links because I still need to sort them and link them to the previous image links.

**Comment by FrutyX** [comment]
Alright then, if you need some help in a way I am able to, you know where to find me.

I'll keep it as WIP status for now.

**Comment by FrutyX** [comment]
Failed to detect enemies in underworld when a secondary drop drops (no costume), but I can also see that UW enemies weren't added yet, so if this is helpful:


**Our avatar (no costume):** https://gf2.geo.gfsrv.net/cdn12/124a63795958541c73a4e72afa1eb1.jpg
**Mercury:** https://gf1.geo.gfsrv.net/cdn67/3754533f396ef58b2a2e6e10ae2ce1.jpg
**Dead Souls:** https://gf1.geo.gfsrv.net/cdn0e/150f321a110eb56d29722525352ab8.jpg
**Ferryman Charon:** https://gf1.geo.gfsrv.net/cdnc2/45b98b77994de23b38c7d7a4c66cc6.jpg
	
**Cerberus:** https://gf3.geo.gfsrv.net/cdn55/23d1b46fc425b94d47901cfa26b617.jpg
**Aeacus:** https://gf1.geo.gfsrv.net/cdn3a/91992511c6e3d3a12395c809a2af6b.jpg
**Rhadamanthus:** https://gf3.geo.gfsrv.net/cdneb/a0bee3578089807e4903068fb7bf78.jpg
**Minos:** https://gf3.geo.gfsrv.net/cdn2d/bf68cbfb4a12f9876f83cd855ba703.jpg
	
**Hydra:** https://gf2.geo.gfsrv.net/cdn1a/abc607343a9b46c6b69173b19f434f.jpg
**Centimanes:** https://gf2.geo.gfsrv.net/cdn40/229a35f7226acf2deb140a59394853.jpg
**Tisiphone:** https://gf1.geo.gfsrv.net/cdn36/46eebfc698733002bdb9d50168fa34.jpg
**Typhon:** https://gf2.geo.gfsrv.net/cdn40/ad32b6084a5cf5178b83d6a0ee3230.jpg
	
**Proserpina:** https://gf2.geo.gfsrv.net/cdnd2/6703d930e67e77674d505a6c2c9d3a.jpg
**Alecto & Megaera:** https://gf3.geo.gfsrv.net/cdn20/45bc06f6e42d5840b4cb077809bced.jpg
**Sphinx:** https://gf2.geo.gfsrv.net/cdn70/739ca8bf810cde7cac6ab78c2da517.jpg
**Dis Pater:** https://gf1.geo.gfsrv.net/cdnf8/78e9dd30a5e0f07d5be66c9bae9dd4.jpg
	
**(Bonus) Unknown:** https://gf2.geo.gfsrv.net/cdnae/904194973d21066c96cb414d04d676.jpg

**Comment by FrutyX** [comment]
The feature, that is supposed to show rubies price in the mystery box price pool is bugged, it appears it's also because of image link:

<details>

![rubies](https://user-images.githubusercontent.com/23278552/180775477-3e968fd9-5119-49e9-a29f-782c72b2a0d0.PNG)
![rubiesprices](https://user-images.githubusercontent.com/23278552/180775482-d565a436-c4c2-45f8-aa84-5ce74ba982d9.PNG)

</details>

**Comment by FrutyX** [comment]
In pantheon - quests, it seems that GCA is trying to load a broken image:

`inner.jpg:1          GET https://cz.gladiatus.gameforge.com/game/img/ui/training/inner.jpg 404 (Not Found)`

**Comment by FrutyX** [comment]
Nevermind, ignore the last message (deleted it), it was something I caused :).

**Comment by GramThanos** [comment]
The only problem that still exists is the fact that some images are the same for multiple NPC enemies, thus from the url of the CDN image, we can not understand which NPM it is. In these cases, the addon assumes that the it is the first of the NPCs.
This may cause problems on the drop statistics but for now, I think these NPCs are not in expedition locations.

I craweld the images from the gamefoge servers, and added all the enemies from the expeditions on the data. Furthermore, following gf approach, I implemented a code to try and map images that the addon is using on the CDN servers to reduce the load on the main game servers. This uses as a fall back the `/cdn/` directory.

So right now you may encounter an error `Image link "{CDN-LINK-HERE}" was not found on CDN lookup list.` which does not break the addon but informs us that this image is missing from our data.
And also you may encounter a warning `Image path "{OLD-IMG-PATH}" was not found on CDN lookup list. Using "{CDN-FOLDER-FALLBACK}".` which just informs us that data to map this image to the CDN server are missing.

**Comment by GramThanos** [comment]
To sum up. Expedition drops should work. On mystery box not all custome drops are detectable (does not affect outcoume). Not all images used from the addon points to the CDN (does not affect the addon's operation).

Thus, if we don't have any other problem, we may start releasing a version.

**Comment by FrutyX** [comment]
So for now, this is expected?

![error3](https://user-images.githubusercontent.com/23278552/183107496-7206a9dc-5040-456e-a017-c4452dfa5c6b.PNG)

**Comment by GramThanos** [comment]
it depends on where you found it.

**Comment by FrutyX** [comment]
I've reinstalled GCA and restarted my browser to be sure, it seems this one is gone now, but I'll keep my eye on it.

**Comment by FrutyX** [comment]
Hmmm, the stuff that is not buyable for rubies, or does not have a ruby value, should probably not have an value shown? Potions are an example.

![value](https://user-images.githubusercontent.com/23278552/183112479-f5ba8603-626c-4221-bf0a-69c17ec11f21.PNG)

**Comment by FrutyX** [comment]
@GramThanos I need something from you, a confirmation, all I need you to do is, to make me a screen of an enemy, but the enemy needs his materials shown, that's all.

**Comment by FrutyX** [comment]
And as he magically appeared, he also dissapeared 🤣 .

**Comment by GramThanos** [comment]
It's beer o'clock mate.

**Comment by GreatApo** [comment]
![PXL_20220805_184206852.jpg](https://user-images.githubusercontent.com/9620510/183141220-b1cfd5af-eafc-4114-87a1-5c1e3399420b.jpg)

**Comment by FrutyX** [comment]
Nice, now that is understandable!

**Comment by GramThanos** [comment]
@FrutyX
![image](https://user-images.githubusercontent.com/14858959/183184094-17e9c508-ebb4-4cf7-ad0d-fd1777396b9b.png)

**Comment by FrutyX** [comment]
Good, thanks.

With those updates and merges, Gameforge also did some changes to their CSS (or just the page layout), and some GCA things are moved, however I wanted an confirmation, because I am using a custom scaling in my Windows, and sometimes it's changing thinks the way they should not look like (bad page rendering, but they did improved it a lot), I do have my ways to test it, but I wanted to be 100% sure.

**Comment by FrutyX** [comment]
Hmm, I think the training page doesn't work, can you do a quick check?

`Uncaught SyntaxError: Unexpected identifier (at training.js?4.3.6&built=1659739158298:220:61)`

**Comment by FrutyX** [comment]
Think I've fixed it, I guess.

**Comment by GramThanos** [comment]
Kind of :P

It was a syntax error, meaning the interpreter (the program that reads and executes the code) was not able to read the file as the JavaScript syntax was not correct.

In particular we had a string, then space, then a function call. You added the semicolon, which fixed the syntax but this was not the intended thing to do. The part that I broke was that I forgot to add a + sign, which means concatenate the string with the function output creating a new string.

Got it?

**Comment by FrutyX** [comment]
Yeah, probably, yes, I do.

:D

**Comment by FrutyX** [comment]
Well, if you don't plan any changes or improvements for the reports and mystery box values, I would say we are pretty much ready to release 4.3.6, however, there is always room for improvements, testing and bug catching, always something to do, and since you don't release a new version very often, then we need to make sure, that this is it.

**Comment by FrutyX** [comment]
I am aware, that you are aware, but if any helpful:

**Not found on CDN lookup list:**

https://gf1.geo.gfsrv.net/cdn37/c4364b8ac67369247214b8f71fd0b5.jpg

**Failed to detect enemy:**

https://gf1.geo.gfsrv.net/cdnf8/78e9dd30a5e0f07d5be66c9bae9dd4.jpg
https://gf3.geo.gfsrv.net/cdn29/bac7e539d61a9de6caec21363c2694.jpg

Looking for errors is pain now, because logged errors are spammed with all these cdn warning.

**Comment by GramThanos** [comment]
> https://gf1.geo.gfsrv.net/cdnf8/78e9dd30a5e0f07d5be66c9bae9dd4.jpg
> https://gf3.geo.gfsrv.net/cdn29/bac7e539d61a9de6caec21363c2694.jpg

These 2 are already in the data.

**Comment by FrutyX** [comment]
And should this "failed" error occur, with data added?

**Comment by GramThanos** [comment]
If you have the last version of the extension I guess no.

**Comment by FrutyX** [comment]
I thought so, and yet, I am still getting these errors.

**Comment by GramThanos** [comment]
> I thought so, and yet, I am still getting these errors.

Both links return a value, thus you shouldn't get an error.
```
gca_tools.img.resolve('https://gf1.geo.gfsrv.net/cdnf8/78e9dd30a5e0f07d5be66c9bae9dd4.jpg')
```
```
gca_tools.img.resolve('https://gf3.geo.gfsrv.net/cdn29/bac7e539d61a9de6caec21363c2694.jpg')
```

**Comment by FrutyX** [comment]
I have my suspicion, I'll be back when I confirm or deny it.

**Comment by FrutyX** [comment]
Confirmed, a secondary drop is causing this errors (very #368 related):

`Failed to detect enemy url("//gf3.geo.gfsrv.net/cdn29/bac7e539d61a9de6caec21363c2694.jpg")`
![scd](https://user-images.githubusercontent.com/23278552/183431391-1cb182af-b19e-4394-b006-acce49f14c26.PNG)

**Comment by FrutyX** [comment]
Even the symbols came back for this one!

![Uploading symbols.PNG…]()

**Comment by FrutyX** [comment]
You need this for the list?

https://gf3.geo.gfsrv.net/cdn8b/84fd3f76ff45db7d2cdbf5680aa09b.png

**Comment by FrutyX** [comment]
Helpful?

https://gf3.geo.gfsrv.net/cdnb8/86e9550484e557ea7b3e0b4a825490.jpg
https://gf2.geo.gfsrv.net/cdn4a/19e794899bda1deaf03b3339a2e6f3.jpg

**Comment by FrutyX** [comment]
> Confirmed, a secondary drop is causing this errors :

And it does look like, that the secondary drop has to be a material, for this error to happen.

**Comment by FrutyX** [comment]
So, a recommended To-Do List before a version launch (only things I am currently aware of):

- Pull request #402 
- Once #402 is done, I still have some CSS improvements ready to implement
- You may want to check the "Failed to detect enemy" (explained above) error
- Last but not least there is the mystery box showing weird ruby values on some items, if you want to check that too

Time to time I get some random error, harmless ones I would say, unfortunately my browser dumped my error logs, probably because they were overspammed, but like I said, they are random and harmless with no priority, I haven't noticed anything that would break the extensions, they can also appear now thanks to the loading times, which is absolutely understandable.

**Comment by FrutyX** [comment]
Nile event enemies:

```
Image link "//gf1.geo.gfsrv.net/cdn65/9f45b7fd7ac900004dcafc2feec57d.jpg" was not found on CDN lookup list.
Image link "//gf2.geo.gfsrv.net/cdn10/f0a5434c803bde1ba0913aecd9751b.jpg" was not found on CDN lookup list.
Image link "//gf2.geo.gfsrv.net/cdnae/23d6d58d2cf752711ec4a580848476.jpg" was not found on CDN lookup list.
Image link "//gf2.geo.gfsrv.net/cdn74/531075909c1e2cb313eda2bc1e8c8e.jpg" was not found on CDN lookup list.
```

**Comment by FrutyX** [comment]
I think I have done everything I wanted to do, everything is currently in the testing phase, I may do some little enhancements, bug fixing and this kind of stuff and the more  time we have for testing, the better, however, the biggest question of all question, when we are going to release the new version? The wolf is already knocking at the door, it's almost about time.

The current problems are mentioned above, but these are mostly between us, things people probably will not even notice.

**Comment by FrutyX** [comment]
I am off to sleep, you take the wheel Thanos, improve but just don't broke anything, I worked hard on it as an javascript amateur :D.

**Comment by FrutyX** [comment]
> * Last but not least there is the mystery box showing weird ruby values on some items, if you want to check that too

Yo what, I just had a look, you mind explaining to me why you put some weird values on it, before I'll edit it so it makes sense?

Like why for example `"hamper" : {rubies : 1.07},` (cervisia) has such value? It's an item that cannot be bought, so you can't put a price tag on it, same goes for 100% healing potion (but we could use the price of Centurion activation, because you get it with it).

Someone has been lazy! :D

**Comment by FrutyX** [comment]
Alright, they have to be placeholders, right?

**Comment by FrutyX** [comment]
Well, I would say that's another thing solved.

**Comment by GramThanos** [comment]
If something can not by bought I suggest to show nothing. We dont have to add more translations.
(In greek the unbuyable is "Δεν μπορεί να αγοραστεί"... I would like to see you fitting that in the column).

Regarding the 1.07:
you can buy 14 days centurion for 15 rubies thus 1.0714285714285714 rubies per day. Thus 1 cervicia (24hours) could be 1.07 rubies.

**Comment by GramThanos** [comment]
Ideally the rubies numbers are for people wanting to understand what the gain is from the rewards and help them assess whether it make sense to refresh or not. Further more one could also aggregate the rubies values on the rewards and open only boxes with high gain. But for this to be done, the user's opinion on what exactly they are interesting in is more important.

**Comment by FrutyX** [comment]
Should've explained this to me 24 or 17 days ago, when I was mentioning it, and not when I make changes to it...

Thanos The Troll.

**Comment by GramThanos** [comment]
Look, I just say my opinion, I am not here to revert changes. If you think this should change maybe it should.

I argue so that we can reach a consensus.
Thus, if you still think they are not needed, we can remove them.

**Comment by FrutyX** [comment]
I have made some changes.

- Mobilization and Gate key cannot be bought, but found in the Chest = 1 for 7 rubies.
- 100% Healing Potion you get get one when activating 14 days Centurion = 1 for 15 rubies.
- Magical Bag can be found in the Chest = 1 for 7 rubies.
- Costumes can be found in the Chest = 1 for 7 rubies.
- Cervisia is back at it's previous value according to the calculations.

**Comment by FrutyX** [comment]
By the way, this is most likely a bug, and costumes are not recognized event with the value, shown as "?".

**Comment by FrutyX** [comment]
`Image link "https://gf3.geo.gfsrv.net/cdnba/fb5aa808eb79bf95e982817b87ff72.jpg" was not found on CDN lookup list.`

But I added it on the list? https://github.com/DinoDevs/GladiatusCrazyAddon/commit/6426fea3e8be3858e4b190afa897ff8c0e44acb1

`'img/costumes/sets/male/{pluto-we-dont-know-this}.jpg' : 'fb5aa808eb79bf95e982817b87ff72', // Pluto`

I tried refreshing the addon.

**Comment by GramThanos** [comment]
There is a `ba` missing from the string.
It was supposed to be:

`'img/costumes/sets/male/{pluto-we-dont-know-this}.jpg' : ba'fb5aa808eb79bf95e982817b87ff72', // Pluto`

**Comment by FrutyX** [comment]
Ah so, didn't know that.

**Comment by GramThanos** [comment]
> Ah so, didn't know that.

i probably forgot to copy the 2 chars next to the '/cdn'

**Comment by FrutyX** [comment]
Well, with this fixed, if this is supposed to show the costume prices, it doesn't work, not related to the cdn link.

```
	// game/assets/img/costumes/sets/male/7_complete_small.jpg
			costumes : {rubies : 7}
		},
```

**Comment by FrutyX** [comment]
Thanos, I would just like to remind you, that people that don't use the dev version (so 99.9% GCA users), are having a broken extension, I hate pushing and annoying you, but it's maybe time... I am not seeing anything with high priority that is a problem, the mystery chest does not show costumes ruby values which is a very small bug and the failed to detect enemy error, well, that is happening only on a secondary material drop, and I've checked the reports and the drop is normally saved, so it's basically only a console warning, but it does not really break the extension in any way.

Other than that, I haven't even detected any other errors, of course there is always still room for improvements, bug fixing and new features, but like this, we would never release a new version.

**Comment by GramThanos** [comment]
Hey. I used the flows I scripted a few months ago to push the versions. I think they run correctly. Lets see.

**Comment by GramThanos** [comment]
Update, I got an email from Firefox that the version is live.

**Comment by FrutyX** [comment]
![kraken](https://user-images.githubusercontent.com/23278552/189730216-b01acbcd-8a21-4572-afbd-c40427f01171.gif)

**Comment by FrutyX** [comment]
It seems it's live on Firefox store.

**Comment by FrutyX** [comment]
Okie, I moved everything relevant to it's own issues, and this seems to be completed.

From what I know from Apo (about Thanos), I will keep the finished issues open for one or two weeks after the release (in case there are any problems) and then close them all, for moderation purposes.

**Comment by GramThanos** [comment]
Purge everything.

**Comment by FrutyX** [comment]
I love purging!

**Comment by FrutyX** [comment]
Patched in v4.3.6, closing, re-open if needed.
